sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -311,6 +311,16 @@ case class FilterEstimation(plan: Filter) extends Logging {
       logDebug("[CBO] No statistics for " + attr)
       return None
     }
+
+    attr.dataType match {
+      case _: NumericType | DateType | TimestampType | BooleanType =>
+        if (!colStatsMap.hasMinMaxStats(attr)) {
```

Comment:
sorry I haven't touched CBO for a while and need more context to warm up.

How can this happen? We checked `if (!colStatsMap.contains(attr))` before, and it weird that a column has statitics but has no basic min/max data.

Reply:
I'm not confident to merge this PR without the answer of this question.

Reply:
> sorry I haven't touched CBO for a while and need more context to warm up.
> 
> How can this happen? We checked `if (!colStatsMap.contains(attr))` before, and it's weird that a column has statistics but has no basic min/max data.

min/max is indeed available, but for me, I don't use min/max at the moment, so I don't store it。
More importantly, based on SPARK-23445 considerations：
> min, max and histogram for columns were optional already. Having them all optional is more consistent......

Reply:
> sorry I haven't touched CBO for a while and need more context to warm up.
> 
> How can this happen? We checked `if (!colStatsMap.contains(attr))` before, and it's weird that a column has statistics but has no basic min/max data.

The predecision of the evaluateBinaryForNumeric() method can explain this。
The method path is evaluateBinary() -> evaluateBinaryForNumeric()

Reply:
> I'm not confident to merge this PR without the answer of this question.

The background is that I am solving the problem of Executor OOM caused by large tables being broadcast through CBO.
Our hive tables are all stored in ORC format, and our HMS does not store the statistics of each column, I tried to get the statistics of each column by reading the ORC Footer. But it is too expensive to count all ORC files, I choose to sample some ORC files to get avgLen, and there is no way to get the exact min/max property in this case.

Reply:
> I'm not confident to merge this PR without the answer of this question.

Although the scenario I encountered is rather specific, but I think this is still a constructive improvement.
The evaluateEquality() method and evaluateBinary() method should be consistent.
For numeric types, evaluateBinary() has three judgments: 1. !colStatsMap.contains(attr)
2. !colStatsMap.hasMinMaxStats(attr)
3. !colStatsMap.hasDistinctCount(attr)
but evaluateEquality() has only one judgment.

Reply:
Do you mean this bug only happens in your Spark fork where you changed the statistics collecting logic to not store min/max?

also cc @wzhfy

Reply:
> Do you mean this bug only happens in your Spark fork where you changed the statistics collecting logic to not store min/max?
> 
> also cc @wzhfy

Yep

Reply:
> Do you mean this bug only happens in your Spark fork where you changed the statistics collecting logic to not store min/max?
> 
> also cc @wzhfy

I checked the community version of Spark's method for collecting statistics and it looks like it is able to get min/max.
However, if the statistics are not collected with min/max, then the community version of Spark will also get unexpected results, as described in the test case I added.